File: apps/ios/Luna Stories/Network/UsageAPI.swift
//
//  UsageAPI.swift
//  Luna Stories
//

import Foundation

/// Mirrors GenerationUsageDTO — weekly quota for one kind of generation.
nonisolated struct GenerationUsage: Decodable {
    let message: String
    let used: Int
    let total: Int
    let remaining: Int
    let percentUsed: Int
    let resetsAt: String
}

/// Mirrors UsageSummaryDTO — both quotas at once (GET /usage).
nonisolated struct UsageSummary: Decodable {
    let stories: GenerationUsage
    let audio: GenerationUsage
}

enum UsageAPI {
    static func fetch() async throws -> UsageSummary {
        try await APIClient.shared.get("/api/v1/usage")
    }
}

File: apps/android/app/src/main/java/com/cortexlumora/lunastories/network/UsageAPI.kt
package com.cortexlumora.lunastories.network

import kotlinx.serialization.Serializable

/** Mirrors GenerationUsageDTO — weekly quota for one kind of generation. */
@Serializable
data class GenerationUsage(
    val message: String,
    val used: Int,
    val total: Int,
    val remaining: Int,
    val percentUsed: Int,
    val resetsAt: String,
)

/** Mirrors UsageSummaryDTO — both quotas at once (GET /usage). */
@Serializable
data class UsageSummary(
    val stories: GenerationUsage,
    val audio: GenerationUsage,
)

/**
 * Weekly usage quotas. Best-effort on the client — a failed fetch keeps the
 * UI quiet rather than surfacing an error, since usage display is non-critical.
 * Mirrors iOS UsageAPI.
 */
object UsageAPI {
    suspend fun fetch(): UsageSummary = APIClient.get("/usage")
}
